t/50file-custom-handler.t

```diff
@@ -0,0 +1,82 @@
+use strict;
+use warnings;
+use Test::More;
+use t::Util;
+
+plan skip_all => 'curl not found'
+    unless prog_exists('curl');
+
+subtest 'cgi file' => sub {
+    eval q{use CGI; 1}
+        or plan skip_all => 'CGI.pm not found';
+
+    # spawn h2o
+    my $server = spawn_h2o(<< "EOT");
+file.custom-handler:
+  extension: .cgi
+  fastcgi.spawn: "exec \$H2O_ROOT/share/h2o/fastcgi-cgi"
+hosts:
+  default:
+    paths:
+      /robots.txt:
+        file.file: @{[DOC_ROOT]}/hello.cgi
+EOT
+    run_with_curl($server, sub {
+        my ($proto, $port, $cmd) = @_;
+        my $resp = `$cmd --silent $proto://127.0.0.1:$port/robots.txt?name=Tobor`;
+        is $resp, "Hello Tobor";
+    });
+};
+
+subtest 'cgi file with no extension' => sub {
+    eval q{use CGI; 1}
+        or plan skip_all => 'CGI.pm not found';
+
+    # spawn h2o
+    my $server = spawn_h2o(<< "EOT");
+file.custom-handler:
+  extension: default
+  fastcgi.spawn: "exec \$H2O_ROOT/share/h2o/fastcgi-cgi"
+hosts:
+  default:
+    paths:
+      /robots.txt:
+        file.file: @{[DOC_ROOT]}/noextcgi
+EOT
+    run_with_curl($server, sub {
+        my ($proto, $port, $cmd) = @_;
+        my $resp = `$cmd --silent $proto://127.0.0.1:$port/robots.txt?name=Tobor`;
+        is $resp, "Hello Tobor";
+    });
+};
+
+subtest 'directory containing cgi files' => sub {
+    eval q{use CGI; 1}
+        or plan skip_all => 'CGI.pm not found';
+
+    # spawn h2o
+    my $server = spawn_h2o(<< "EOT");
+file.custom-handler:
+  extension: ["default", ".cgi"]
+  fastcgi.spawn: "exec \$H2O_ROOT/share/h2o/fastcgi-cgi"
```

Comment:
Could we use possibly do `file.mime.settypes: {}` as well to reset the existing mime-type here, and see that a file with `.html` extension gets treated as a CGI?